caffe2/python/modeling/initializers.py

```diff
@@ -29,6 +29,33 @@ def create_param(self, param_name, init_net, shape):
             shape=shape,
         )
 
+class pFP16Initializer(object):
+    def __init__(self, operator_name=None, **kwargs):
+        self.operator_name = operator_name
+        self.operator_kwargs = kwargs
+
+    def update(self, operator_name, kwargs):
+        if self.operator_name is not None:
+            raise Exception("Operator name overwrites are not allowed")
+        self.operator_name = operator_name
+        self.operator_kwargs = kwargs
+
+    def create_param(self, param_name, init_net, shape):
+        # create master fp32 copy
+        param_fp32 = init_net.__getattr__(self.operator_name)(
+            [], param_name+'_fp32', shape=shape,
+            **self.operator_kwargs)
+        # cast to fp16 copy
+        param = init_net.FloatToHalf(
+            param_name+"_fp32", param_name)
+
+        return ParameterInfo(
+            param_id=None,
+            param=param,
+            shape=shape,
+            blob_copy={'float' : param_fp32}
```

Comment:
Doesn't have to be, but it makes sense to me that it is a dict - you also need to know what the different multi-precision types are when updating parameters, so this solves that problem too.